svg.js.d.ts

```diff
@@ -144,46 +614,421 @@ declare module "@svgdotjs/svg.js" {
+        element(element: string, inherit?: Object): this;
+        first(): Element;
+        get(i: number): Element;
+        getEventHolder(): LinkedHTMLElement;
+        getEventTarget(): LinkedHTMLElement;
+        has(element: Element): boolean;
+        id(): string
+        id(id: string): this
+        index(element: Element): number;
+        last(): Element;
+        matches(selector: string): boolean;
+        /**
+         * Get the parent of current element. The input query can be given with string, object type or none (undefined).
+         * The input is vary based on the implement in hierarchy of SVG.JS element or dom.
+         * 1. If The input is a string, the string value must be a valid HTML element tag name or svg tag name. e.g "svg" or "g" or "div"
+         * 2. If the given input is an object type then only SVG.JS object type is accept. e.g Dom, Svg or G
+         * 3. if the given input query is undefined then the element will return the closest parent in Dom hierarchy
+         *
+         * For more information see ParentQueryMapping.
+         * @param type can be either string, object type or undefined.
+         */
+        parent<T extends this>(type?: ParentQueryMapping<T>): ParentQueryResultMapping<T>;
+        put(element: Element, i?: number): Element;
+        /**
+         * Put the element into the given parent element. The input parent element can be vary base on the class implementation.
+         * 1. If the current class is a Dom then parent input is only accept a valid HTML element or a valid string id of HTML element
+         * 2. If the current class is an Svg then parent input can only Dom, Svg, G, A, a valid HTML element and a valid string id of HTML element
+         *
+         * For more information see PutInMapping.
+         * @param parent an object of SVG.JS Dom or implement container or a string id or a valid HTML element object.
+         */
+        putIn<T extends this>(parent: PutInMapping<T>): PutInResultMapping<T>;
+
+        remove(): this;
+        removeElement(element: Element): this;
+        replace<T extends Dom>(element: T): T;
+        round(precision?: number, map?: string[]): this
+        svg(): string;
+        svg(a: boolean, outer?: boolean): string
+        svg(a: null | Function, outer?: boolean): string
+        svg(a: string, outer?: boolean): string
+
         toString(): string;
-        toHex(): string;
-        toRgb(): string;
-        brightness(): number;
-        morph(color: ColorAlias): Color;
-        at(pos: number): Color;
+        words(text: string): this;
+        writeDataToDom(): this;
+
+        // prototype extend Attribute in attr.js
+        /**
+         * Get the attribute object of SVG Element. The return object will be vary based on
+         * the instance itself. For example, G element will only return GlobalAttr where Rect
+         * will return RectAttr instead.
+         */

```

Comment:
This seems like this becomes super complicated. What about custom attributes and data-attributes? Can you cover all this? I mean - I dont mind as long as this works :)

Reply:
I think he gave up and just returned "any"

Reply:
Well thats fine for now I guess :D